src/main/java/teammates/ui/webapi/action/SearchLinksAction.java

```diff
@@ -0,0 +1,172 @@
+                courseIds.add(instructor.courseId);
+            }
+        }
+    }
+
+    private void populateCourseIdToInstituteMap() {
+        for (String courseId : courseIds) {
+            String instructorForCourseGoogleId = findAvailableInstructorGoogleIdForCourse(courseId);
+            AccountAttributes account = logic.getAccount(instructorForCourseGoogleId);
+            if (account == null) {
+                continue;
+            }
+
+            String institute = StringHelper.isEmpty(account.institute) ? "None" : account.institute;
+            courseIdToInstituteMap.put(courseId, institute);
+        }
+    }
+
+    /**
+     * Finds the googleId of a registered instructor with co-owner privileges.
+     * If there is no such instructor, finds the googleId of a registered
+     * instructor with the privilege to modify instructors.
+     *
+     * @param courseId
+     *            the ID of the course
+     * @return the googleId of a suitable instructor if found, otherwise an
+     *         empty string
+     */
+    private String findAvailableInstructorGoogleIdForCourse(String courseId) {
+        List<InstructorAttributes> instructorList = logic.getInstructorsForCourse(courseId);
+
+        for (InstructorAttributes instructor : instructorList) {
+            if (instructor.isRegistered()
+                    && (instructor.hasCoownerPrivileges()
+                    || instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR))) {
+                courseIdToInstructorGoogleIdMap.put(courseId, instructor.googleId);
+                return instructor.googleId;
+            }
+
+        }
+
+        return "";
+    }
+
+    private List<SearchLinksStudentData> getStudentsBundle(List<StudentAttributes> students) {
+        List<SearchLinksStudentData> studentsBundle = new ArrayList<>();
+
+        for (StudentAttributes student : students) {
+            SearchLinksStudentData studentData = new SearchLinksStudentData();
+
+            if (student.email != null && student.course != null
+                    && !StringHelper.isEmpty(courseIdToInstructorGoogleIdMap.get(student.course))) {
+
+                studentData.setRecordsPageLink(Config.getFrontEndAppUrl(Const.WebPageURIs.INSTRUCTOR_STUDENT_RECORDS_PAGE)
+                        .withCourseId(student.course)
+                        .withStudentEmail(student.email)
+                        .withUserId(courseIdToInstructorGoogleIdMap.get(student.course))
+                        .toAbsoluteString());
+            }
+
+            if (student.email != null) {
+                studentData.setEmail(student.email);
+                studentData.setManageAccountLink(Config.getFrontEndAppUrl(Const.WebPageURIs.ADMIN_ACCOUNTS_PAGE)
+                        .withInstructorId(student.googleId).toString());
+                studentData.setHomePageLink(Config.getFrontEndAppUrl(Const.WebPageURIs.INSTRUCTOR_STUDENT_RECORDS_PAGE)
```

Comment:
This doesn't look like the home page link, is it a typo? 🤔

Reply:
👀 good catch thanks